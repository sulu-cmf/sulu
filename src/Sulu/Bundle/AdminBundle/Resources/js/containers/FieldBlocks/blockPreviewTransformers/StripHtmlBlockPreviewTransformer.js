// @flow
import React from 'react';
import type {Node} from 'react';
import textVersion from 'textversionjs';
import type {BlockPreviewTransformer} from '../types';

const MAX_LENGTH = 500;

export default class StripHtmlBlockPreviewTransformer implements BlockPreviewTransformer {
    transform(value: *): Node {
        if (typeof value !== 'string') {
            return null;
        }

        const text = textVersion(value, {headingStyle: 'linebreak', listStyle: 'linebreak'});

        return (
            <p>
                {text.length > MAX_LENGTH ? text.substring(0, MAX_LENGTH) + '...' : text}
            </p>
        );
    }
}
