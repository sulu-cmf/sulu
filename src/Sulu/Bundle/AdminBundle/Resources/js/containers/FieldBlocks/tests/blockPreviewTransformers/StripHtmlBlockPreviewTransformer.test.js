// @flow
import StripHtmlBlockPreviewTransformer from '../../blockPreviewTransformers/StripHtmlBlockPreviewTransformer';

test('Return JSX for simple string', () => {
    const stripHtmlBlockPreviewTransformer = new StripHtmlBlockPreviewTransformer();
    expect(stripHtmlBlockPreviewTransformer.transform('<strong>Test</strong>')).toMatchSnapshot();
});

test('Return JSX for simple string', () => {
    const stripHtmlBlockPreviewTransformer = new StripHtmlBlockPreviewTransformer();
    expect(stripHtmlBlockPreviewTransformer.transform('<strong>' + 'c'.repeat(1000) + '</strong>')).toMatchSnapshot();
});

test('Return null for everything expect a string', () => {
    const stripHtmlBlockPreviewTransformer = new StripHtmlBlockPreviewTransformer();
    expect(stripHtmlBlockPreviewTransformer.transform({})).toMatchSnapshot();
});
