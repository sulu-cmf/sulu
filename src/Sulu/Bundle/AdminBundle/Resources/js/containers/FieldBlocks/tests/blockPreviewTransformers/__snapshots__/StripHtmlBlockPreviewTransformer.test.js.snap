// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return JSX for simple string 1`] = `
<p>
  Test

</p>
`;

exports[`Return JSX for simple string 2`] = `
<p>
  cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc...
</p>
`;

exports[`Return null for everything expect a string 1`] = `null`;
